File: 3553-check-if-two-chessboard-squares-have-the-same-color/check-if-two-chessboard-squares-have-the-same-color.py
class Solution:
    def checkTwoChessboards(self, coordinate1: str, coordinate2: str) -> bool:
        return self._is_black(coordinate1) == self._is_black(coordinate2)

    def _is_black(self, coord: str) -> bool:
        file_num = ord(coord[0]) - ord('a') + 1
        rank = int(coord[1])
        return ((file_num + rank) % 2) == 0

File: 3553-check-if-two-chessboard-squares-have-the-same-color/check-if-two-chessboard-squares-have-the-same-color.java
class Solution {
    public boolean checkTwoChessboards(String coordinate1, String coordinate2) {
        return isBlack(coordinate1) == isBlack(coordinate2);
    }

    private boolean isBlack(String coord) {
        int file = coord.charAt(0) - 'a' + 1;
        int rank = coord.charAt(1) - '0';
        return ((file + rank) & 1) == 0;
    }
}
